aspnetcore/signalr/streaming.md

```diff
@@ -110,29 +160,67 @@ Console.WriteLine("Streaming completed");
 
 ::: moniker-end
 
+::: moniker range=">= aspnetcore-3.0"
+
+### Client to server streaming
+
+To invoke a client to server streaming hub method from the .NET Client create a `Channel` and pass the `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method being invoked.
+
+Whenever data is written to the `ChannelWriter` the hub method on the server will receive a new item with the data from the client.
+
+To end the stream, complete the channel with `channel.Writer.TryComplete()`.
```

Comment:
Why recommend TryComplete() over Complete() here?

Reply:
Same reason we generally use `TrySetResult` and friends. `Complete` will throw if the channel is already completed.

Reply:
I prefer having the Channel throw in simple cases like these. No other code has access to the Writer, so double completion would never be expected and would always be a bug.

Reply:
@BrennanConroy You asked for feedback. I stand by my last comment here.

Reply:
If we weren't meant to use `ChannelWriter.Complete()` in cases like these, the method wouldn't exist.

Reply:
I thought I changed this, must have forgotten.